### PR TITLE
Rework - Sensors build test

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by galak** [comment]
This is first step in getting improved coverage on the sensors.  Want to get this merged, than will look at getting ALL sensor drivers actually building as part of this test.

File: tests/drivers/build_all/sensor/prj.conf
Comment (nashif) [inline tests/drivers/build_all/sensor/prj.conf:1]: where can I buy this board with so many sensors? :-D

**Review by nashif — APPROVED** [review]

**Comment by avisconti** [comment]
> This is first step in getting improved coverage on the sensors. Want to get this merged, than will look at getting ALL sensor drivers actually building as part of this test.

I'll take a closer look tomorrow (Monday). It looks somehow interesting...

**Review by avisconti — APPROVED** [review]
nice idea. Not complete yet. I approve to let you proceed further, as you suggested.

**Comment by avisconti** [comment]
I already approved, but what happens if we merge it like this?

**Review by MaureenHelm — APPROVED** [review]

**Comment by MaureenHelm** [comment]
> I already approved, but what happens if we merge it like this?

I don't understand your question. Are you ok with this getting merged?

**Comment by avisconti** [comment]
> > I already approved, but what happens if we merge it like this?
> 
> I don't understand your question. Are you ok with this getting merged?

I am, but I would like @galak to comment more on this statement:

``
This is first step in getting improved coverage on the sensors. Want to get this merged, than will look at getting ALL sensor drivers actually building as part of this test. ``

Does it mean that after merging the CI will fail until job will be completed? I think it is just my misunderstanding, that's why I still approve. But maybe some more clarification is required here.

File: drivers/sensor/iis2iclx/iis2iclx_trigger.c
Comment (galak) [reply drivers/sensor/iis2iclx/iis2iclx_trigger.c:146]: Yes, there was a warning/error in clang with this.

**Comment by galak** [comment]
> Does it mean that after merging the CI will fail until job will be completed? I think it is just my misunderstanding, that's why I still approve. But maybe some more clarification is required here.

CI should still pass with this change.  My comment was that I wanted this merged before making a number of changes on top of this as I expect we'll starting see more failures with sensors not building in different configs.

The follow PR(s) will do:
* Add all current sensors to build all - will start with `sensors.build`
* Add configs for the 3 common configs for trigger mode - None, global thread, and own thread.

**Pull request merged**
